### PR TITLE
[UI] [Serve Replica] Making the log link works 

### PR DESCRIPTION
**Pull request by chaowanggg**
## Why are these changes needed?
1. Perviously, the log link could not jump to the task page.
2. Now, add a React Route in ServeReplicaDetailPage to make the log link jumping works

https://github.com/ray-project/ray/assets/125417081/625adc0a-7e20-46ee-ab07-9fc23ed4653c

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
LGTM. Thanks!

File: dashboard/client/src/App.tsx [hunk 1]
```diff
@@ -243,6 +243,10 @@ const App = () => {
                       element={<ServeReplicaDetailPage />}
                       path=":deploymentName/:replicaId"
                     />
+                    <Route
+                      element={<TaskPage />}
+                      path=":deploymentName/:replicaId/tasks/:taskId"
+                    />
```
Comment (alanwguo) [inline dashboard/client/src/App.tsx:249]: Can we make this a child route of `<Route
                      element={<ServeReplicaDetailPage />}
                      path=":deploymentName/:replicaId"
                    />`? 
                    
...so the breadcrumbs work correctly.
You may need to do something similar to `ActorDetailLayout`, `ActorDetailPage`, `TaskPage` in line 221.

Comment (chaowanggg) [reply dashboard/client/src/App.tsx]: Updated, please have a look at the recording above

**Review by alanwguo — APPROVED** [review]
lgtm!

**Comment by chaowanggg** [comment]
@scottsun94 
The link of serveReplica ID also works

https://github.com/ray-project/ray/assets/125417081/aefaff50-b260-4b79-b86b-d1896a17c028

**Comment by scottsun94** [comment]
Awesome!

**Pull request merged**